### PR TITLE
Add a .editorconfig file to enforce coding style

### PR DESCRIPTION
**Pull request by wmcmurray**
In case you didn't know :man_shrugging:  , this will automatically set almost any text editor to the proper config :
**utf-8, 2 tabs, LF newlines and no trailing white spaces** ([from Mr.doob's Code Style](https://github.com/mrdoob/three.js/wiki/Mr.doob%27s-Code-Style%E2%84%A2#general-provisions))

It makes things much more easier ! And different config can be applied to different files types, etc.

You just need the plugin for your text editor : https://editorconfig.org/ (there is one for Atom :atom:  )

**If this gets merged**, I guess it could be mentioned in the [wiki page](https://github.com/mrdoob/three.js/wiki/How-to-contribute-to-three.js) before last point.

### CODE REVIEW

**Comment by mrdoob** [comment]
Hmm, is very rare these days to see Pull Requests that don't follow that code style.
Makes me wonder if this is really needed...?

**Comment by gkjohnson** [comment]
@mrdoob FWIW I have an uncommitted `.editorconfig` and `.eslintrc.json`  file (that uses the `eslint-config-mdcs` config) in my local three.js repo to ensure that code fits the expected style.  I think maintaining these files along side the main repo will make it easier to track if the code style ever  changes in the future (switching from var to let / const for example).

I would vote for adding this and an eslint file to the repo.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[*.{js,ts,html}]
+charset = utf-8
+indent_style = tab
+indent_size = 2
```
Comment (gkjohnson) [inline .editorconfig]: If this is merged I don't think the config should enforce indent width considering it is only a visual configuration and doesn't affect the file written to disk. I often use 4 columns to indent, for example.

**Comment by wmcmurray** [comment]
@mrdoob well it's not "needed", but it puts a burden aside for every person wanting to contribute :man_shrugging: 

...and it's pretty common : [vuejs](https://github.com/vuejs/vue/blob/master/.editorconfig), [react](https://github.com/facebook/react/blob/master/.editorconfig), [angular](https://github.com/angular/angular.js/blob/master/.editorconfig), [laravel](https://github.com/laravel/laravel/blob/master/.editorconfig), [php](https://github.com/php/php-src/blob/master/.editorconfig), [nodejs](https://github.com/nodejs/node/blob/master/.editorconfig) , name it !

File: .editorconfig (hunk 1 above)
Comment (wmcmurray) [reply .editorconfig]: Sure, don't hesitate to tweak it to your needs, I just wanted to start the debate..
You guys know this codebase much more than I do ! :bowing_man:

**Comment by mrdoob** [comment]
I see I see. Okay!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**